### ISSUE TITLE
How do I ignore internal superclasses?

### ISSUE THREAD

**Issue by Togrias** [ExpediaDotCom/graphql-kotlin#135]
The schema generator should ignore internal superclasses.

Take the following example:

```
inner class User_WidgetsConnection (
  override val edges: User_WidgetsEdge,
  override val pageInfo: PageInfo
): KConnection<Widgets?>
```

`internal interface KConnection<T> (val edges ...)`

will throw: java.lang.IllegalArgumentException: Class declares 1 type parameters, but 0 were provided.

I think the reason is that Graphql does not support <T> typing (which graphql-kotlin correctly complains), but I think the correct behaviour is to ignore the internal KConnection<T> class in the first place.

**Comment by dkuc84** [comment]
Thanks for reporting. Your problem is indeed caused by generics which as you noted is because GraphQL does not support it.

In general, `graphql-kotlin` will attempt to generate schema from the objects that you specify as queries/mutations. Currently we only explicitly filter class properties and functions to only pick up ones with `public` modifier. As it is right now we don't do any filtering on objects and interfaces. Our assumption was that generally they would have public visibility but nothing stops you from defining them as `internal` or even `private`. I'll make a change to make this behavior consistent, i.e. require queries/mutations to be public and only expose public interfaces.

**Comment by smyrick** [comment]
Resolved in https://github.com/ExpediaDotCom/graphql-kotlin/issues/137

**Comment by smyrick** [comment]
@Togrias This was released in `0.0.31`
